nav2_bringup/bringup/launch/tb3_simulation_launch.py

```diff
@@ -149,9 +151,21 @@ def generate_launch_description():
         name='robot_state_publisher',
         namespace=namespace,
         output='screen',
-        parameters=[{'use_sim_time': use_sim_time}],
-        remappings=remappings,
-        arguments=[urdf])
+        parameters=[{'use_sim_time': use_sim_time,
+                     'robot_description': robot_description}],
+        remappings=remappings)
+
+    start_gazebo_spawner_cmd = Node(
+        package='nav2_gazebo_spawner',
```

Comment:
Question: can you use the gazebo spawn model instead? https://github.com/ros-simulation/gazebo_ros_pkgs/blob/231a7219b36b8a6cdd100b59f66a3df2955df787/gazebo_ros/scripts/spawn_model Can you find the major differences between the Nav2 one and the Gazebo one? I believe we made that before Gazebo had that feature and we ported it from ROS 1 so it very well might be the same or so similar we  can remove that. Less code here is always good news

Reply:
sure we can use. there is no major difference between them. it only additionally checks is the model has differential drive.

Reply:
OK sweet, just remove the nav2 version then and replace with gazebo ros's! That's great to hear!!

Reply:
i tried it. it only works without namespace.  tb_differential_drive plugin sends tf to /tf even with namespace. i tried to remap it using remap rule it didn't work out.
![Screenshot from 2021-07-29 22-41-56](https://user-images.githubusercontent.com/57533054/127536550-42660f38-847a-479e-be00-ee1088a9b0c9.png)
i also tried to  bring the block of nav2_spawner code that remaps it to launch,  namespace being runtime variable it doesn't work.
do you have any suggestion to make it work.

Reply:
Did you try the actual nav2_spawner and not just bringing back in the block of code? 

If they both fail, then I don't know what to tell you, its probably not relevant to this PR. Sounds like a ticket to file with the turtlebot3 repo. Did this used to work when TB3 used the ROS2 diff drive plugin? Did you try simply replacing theirs with the generic one? I think that's what we use in the Nav2 version of the TB3 SDF file in https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/bringup/worlds/waffle.model#L516

There's many reasons we do NOT use the TB3 supplied files and why long term we are likely to entirely stop supporting it in Nav2. Its just a minefield of lazy engineering and bad user support

Reply:
nav2_gazebo_spawner works fine. we can finish this PR using it. you earlier said that to try spawn with spawn_entity.py of gazebo_ros. it doesn't work when we spawn with namespace. it has problem with diff_drive_node it remap cmd_vel and odom topic with robot1 namespace but not tf see the above fig. nav2_gazebo_spawner avoid that by https://github.com/ros-planning/navigation2/blob/e96450db3a425d8eb54737dffc09eef0bb1e1391/nav2_bringup/nav2_gazebo_spawner/nav2_gazebo_spawner/nav2_gazebo_spawner.py#L87

nav2_gazebo_spawner stays for now. 
shall i finish this pr i have some changes in rviz files and param file

Reply:
OK makes sense!